model/src/main/kotlin/Provenance.kt

```diff
@@ -68,6 +71,11 @@ data class Provenance(
         }
     }
 
+    /**
+     * Calculate the SHA-1 hash of the string representation of this [Provenance] instance.
+     */
+    fun hash(): String = MessageDigest.getInstance("SHA-1").digest(toString().toByteArray()).toHexString()
```

Comment:
Do you think it makes sense to move `MessageDigest.getInstance("SHA-1")` to a (lazy) private class property now?

Reply:
Good idea, done.